internal/cephfs/controllerserver.go

```diff
@@ -823,6 +819,41 @@ func (cs *ControllerServer) CreateSnapshot(
 	// too.
 	volClient := core.NewSubVolume(parentVolOptions.GetConnection(), &parentVolOptions.SubVolume,
 		parentVolOptions.ClusterID, cs.ClusterName, cs.SetMetadata)
+
+	if parentVolOptions.BackingSnapshot {
+		// update the reference count for parent volume
+		if err = store.UpdateRefCount(ctx, parentVolOptions, cs.ClusterName,
+			cs.SetMetadata, req.GetSecrets()); err != nil {
+			return nil, status.Error(codes.InvalidArgument, "failed to snapshot a snapshot-backed volume")
+		}
+		// Reservation
+		sID, er := store.ReserveSnap(ctx, parentVolOptions, vid.FsSubvolName, cephfsSnap, cr)
+		if er != nil {
+			return nil, status.Error(codes.Internal, er.Error())
+		}
+
+		defer func() {
+			if er != nil {
+				errDefer := store.UndoSnapReservation(ctx, parentVolOptions, *sID, snapName, cr)
+				if errDefer != nil {
+					log.WarningLog(ctx, "failed undoing reservation of snapshot: %s (%s)",
+						requestName, errDefer)
+				}
+			}
+		}()
+
+		// setting creation time as the current time
+		ts := timestamppb.Now()
```

Comment:
we should actually set the snapshottime here not the current time

Reply:
Thanks Madhu for the review,
So this was done for example - the user creates a snapshot and then the snapshot of pvc-restore a week later, then in snapshot we would like to have the time as a current and not a week before?

Reply:
Please take a look at https://github.com/container-storage-interface/spec/blob/b01039c563108173c6743aa1410ec11fde7c24fe/csi.proto#L1191-L1193. Please note that the timestamp on the snapshot reflects the time it was created on the storage system, not the time it was created by the user.